app/src/common/shared/org/mozilla/vrbrowser/SessionStore.java

```diff
@@ -77,10 +73,58 @@ public static SessionStore get() {
         GeckoSession mSession;
     }
 
+    private static class GeckoError {
+
+        public static final String ERROR_URL = "resource://android/assets/html/error/netError.xhtml";
+
+        public static final String NET_ERROR_TYPE = "about:neterror";
+        public static final String CERT_ERROR_TYPE = "about:certerror";
```

Comment:
the error page for https://expired.badssl.com/ should contain different text, including the hostname (`expired.badssl.com`) being displayed in the text:

![expired badssl com](https://user-images.githubusercontent.com/203725/43291299-c8b50c2e-90e5-11e8-8089-24e7e15bf59b.png)
![image](https://user-images.githubusercontent.com/203725/43291373-037366b2-90e6-11e8-95d8-87a46fb01395.png)

> The owner of expired.badssl.com has configured their website improperly. To protect your information from being stolen, Firefox has not connected to this website.

Reply:
same for https://wrong.host.badssl.com/:

![wrong host badssl com 2](https://user-images.githubusercontent.com/203725/43291465-66849ba4-90e6-11e8-9ad4-a7a24e330957.png)

![image](https://user-images.githubusercontent.com/203725/43291438-52208d12-90e6-11e8-81b4-9cd577209693.png)
![image](https://user-images.githubusercontent.com/203725/43291445-568afdb0-90e6-11e8-851e-faf8d6bff118.png)
![image](https://user-images.githubusercontent.com/203725/43291448-589873ee-90e6-11e8-9927-1f7146d80c3b.png)
![image](https://user-images.githubusercontent.com/203725/43291449-5b1915c4-90e6-11e8-99ac-d2645d3551b2.png)

Reply:
For all the SSL error Gecko seems to be reporting all them as **nssBadCert** with the exception of the pinning-test which is reported as **nssFailure2**:

https://expired.badssl.com/
about:certerror?e=nssBadCert&u=https%3A//expired.badssl.com/&c=UTF-8&f=regular&d=%20

https://wrong.host.badssl.com/
about:certerror?e=nssBadCert&u=https%3A//wrong.host.badssl.com/&c=UTF-8&f=regular&d=%20

https://self-signed.badssl.com/
about:certerror?e=nssBadCert&u=https%3A//self-signed.badssl.com/&c=UTF-8&f=regular&d=%20

https://untrusted-root.badssl.com
about:certerror?e=nssBadCert&u=https%3A//untrusted-root.badssl.com/&c=UTF-8&f=regular&d=%20

https://revoked.badssl.com
No error

https://pinning-test.badssl.com
about:neterror?e=nssFailure2&u=https%3A//pinning-test.badssl.com/&c=UTF-8&f=regular&d=%20

I guess this should be addressed also in Gecko.